pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -0,0 +1,476 @@
+		case int, bool, string:
+			args = append(args, fmt.Sprint(typedValue))
+		case []string:
+			args = append(args, strings.Join(typedValue, "|"))
+		default:
+			return errors.Errorf("%v has a non-supported type value: %v", field, value)
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return tx.StmtExec(tx.Stmt(l.addField), args...)
+}
+
+func (l *ListOptionIndexer) afterDelete(key string, tx db.TXClient) error {
+	args := []any{key}
+	return tx.StmtExec(tx.Stmt(l.deleteField), args...)
+}
+
+// ListByOptions returns objects according to the specified list options and partitions
+// result is an unstructured.UnstructuredList, the continue token for the next page (or an error)
+func (l *ListOptionIndexer) ListByOptions(ctx context.Context, lo ListOptions, partitions []partition.Partition, namespace string) (*unstructured.UnstructuredList, string, error) {
+	// 1- Intro: SELECT and JOIN clauses
+	stmt := fmt.Sprintf(`SELECT o.object, o.objectnonce, o.dek, o.deknonce FROM "%s" o`, l.GetName())
+	stmt += "\n  "
+	stmt += fmt.Sprintf(`JOIN db2."%s_fields" f ON o.key = f.key`, l.GetName())
+	params := []any{}
+
+	// 2- Filtering: WHERE clauses (from lo.Filters)
+	whereClauses := []string{}
+	for _, orFilters := range lo.Filters {
+		orClause, orParams, err := l.buildORClauseFromFilters(orFilters)
+		if err != nil {
+			return nil, "", err
+		}
+		whereClauses = append(whereClauses, orClause)
+		params = append(params, orParams...)
+	}
+
+	// WHERE clauses (from namespace)
+	if namespace != "" && namespace != "*" {
+		whereClauses = append(whereClauses, fmt.Sprintf(`f."metadata.namespace" = ?`))
+		params = append(params, namespace)
+	}
+
+	// WHERE clauses (from partitions and their corresponding parameters)
+	partitionClauses := []string{}
+	for _, partition := range partitions {
+		if partition.Passthrough {
+			// nothing to do, no extra filtering to apply by definition
+		} else {
+			singlePartitionClauses := []string{}
+
+			// filter by namespace
+			if partition.Namespace != "" && partition.Namespace != "*" {
+				singlePartitionClauses = append(singlePartitionClauses, fmt.Sprintf(`f."metadata.namespace" = ?`))
+				params = append(params, partition.Namespace)
+			}
+
+			// optionally filter by names
+			if !partition.All {
+				names := partition.Names
+
+				if len(names) == 0 {
+					// degenerate case, there will be no results
+					singlePartitionClauses = append(singlePartitionClauses, "FALSE")
+				} else {
+					singlePartitionClauses = append(singlePartitionClauses, fmt.Sprintf(`f."metadata.name" IN (?%s)`, strings.Repeat(", ?", len(partition.Names)-1)))
+					for name := range partition.Names {
+						params = append(params, name)
+					}
+				}
+			}
+
+			if len(singlePartitionClauses) > 0 {
+				partitionClauses = append(partitionClauses, strings.Join(singlePartitionClauses, " AND "))
+			}
+		}
+	}
+	if len(partitions) == 0 {
+		// degenerate case, there will be no results
+		whereClauses = append(whereClauses, "FALSE")
```

Comment:
@MbolotSuse this case is actually different and `return toUnstructuredList([]any{}), nil, nil` would be valid here, IIUC